include/operations/Operation.hpp

```diff
@@ -135,6 +135,8 @@ class Operation {
   }
 
   virtual void addDepthContribution(std::vector<std::size_t>& depths) const;
+  virtual void
+  addTwoQubitDepthContribution(std::vector<std::size_t>& depths) const;
```

Comment:
This feature doesn't work correctly for a `CompoundOperation`. For these the method would have to be overridden so as to recursively compute the depth.